File: single-language-analysis/dataset-files/python/test_smell_110.py
# Original URL: https://github.com/m-bo-one/ethereumd-proxy/blob/1d1eb3905dac4b28a8e23c283214859a13f6e020/tests/test_poller.py#L126-L155

class TestIncorrectFakePoller(FakePoller):

    @pytest.mark.asyncio
    @setup_proxies
    async def test_smell_110(self):
        with patch('ethereumd.poller.Poller.poll'):
            poller = Poller(self.rpc_proxy)
            assert poller.has_blocknotify is False
            assert poller.has_walletnotify is False
            assert poller.has_alertnotify is False

            poller = Poller(self.rpc_proxy, cmds={'blocknotify': 'echo "%s"'})
            assert poller.has_blocknotify is True
            assert poller.has_walletnotify is False
            assert poller.has_alertnotify is False

            poller = Poller(self.rpc_proxy, cmds={'walletnotify': 'echo "%s"'})
            assert poller.has_blocknotify is False
            assert poller.has_walletnotify is True
            assert poller.has_alertnotify is False

            poller = Poller(self.rpc_proxy, cmds={'alertnotify': 'echo "%s"'})
            assert poller.has_blocknotify is False
            assert poller.has_walletnotify is False
            assert poller.has_alertnotify is True

            poller = Poller(self.rpc_proxy, cmds={'blocknotify': 'echo "%s"',
                                                  'walletnotify': 'echo "%s"',
                                                  'alertnotify': 'echo "%s"'})
            assert poller.has_blocknotify is True
            assert poller.has_walletnotify is True
            assert poller.has_alertnotify is True

File: single-language-analysis/dataset-files/java/TestSmell110.java
import org.junit.Test;
import static org.junit.Assert.*;
import static org.mockito.Mockito.*;

public class TestSmell110 {

    @Test
    public void testMethodHasCommand() {
        try {
            Poller poller = mock(Poller.class);
            when(poller.poll()).thenReturn(null);
            
            poller = new Poller(this.rpcProxy);
            assertEquals(false, poller.hasBlocknotify());
            assertEquals(false, poller.hasWalletnotify());
            assertEquals(false, poller.hasAlertnotify());

            poller = new Poller(this.rpcProxy, new HashMap<String, String>() {{
                put("blocknotify", "echo \"%s\"");
            }});
            assertEquals(true, poller.hasBlocknotify());
            assertEquals(false, poller.hasWalletnotify());
            assertEquals(false, poller.hasAlertnotify());

            poller = new Poller(this.rpcProxy, new HashMap<String, String>() {{
                put("walletnotify", "echo \"%s\"");
            }});
            assertEquals(false, poller.hasBlocknotify());
            assertEquals(true, poller.hasWalletnotify());
            assertEquals(false, poller.hasAlertnotify());

            poller = new Poller(this.rpcProxy, new HashMap<String, String>() {{
                put("alertnotify", "echo \"%s\"");
            }});
            assertEquals(false, poller.hasBlocknotify());
            assertEquals(false, poller.hasWalletnotify());
            assertEquals(true, poller.hasAlertnotify());

            poller = new Poller(this.rpcProxy, new HashMap<String, String>() {{
                put("blocknotify", "echo \"%s\"");
                put("walletnotify", "echo \"%s\"");
                put("alertnotify", "echo \"%s\"");
            }});
            assertEquals(true, poller.hasBlocknotify());
            assertEquals(true, poller.hasWalletnotify());
            assertEquals(true, poller.hasAlertnotify());
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}